### DISCUSSION TITLE
AWS Postgres script deployments with github actions

### DISCUSSION THREAD

**Discussion by a1saipavan** [community/community#32104] [Actions]
Hi There,

I'm trying to execute postgres scripts with github actions, I'm unable to complete this task. can any one please help me on this.

![image](https://user-images.githubusercontent.com/76589153/188581763-0d25b8fe-f7a3-4d45-9701-92cc4ab1afb2.png)

Run psql -h database-1.cnueudck5tkx.us-east-1.rds.amazonaws.com -U postgres -p 5432 -d core -f /__w/postgresdeploy/postgres/Migrations/scripts/postgres.sql
psql: error: could not connect to server: Connection timed out
	Is the server running on host "database-1.cnueudck5tkx.us-east-1.rds.amazonaws.com" (172.31.21.1[6](https://github.com/a1saipavan/postgresdeploy/runs/8202149770?check_suite_focus=true#step:4:7)) and accepting
	TCP/IP connections on port 5432?
Error: Process completed with exit code 2.

Could anyone help me with this? Thanks a lot!

**Comment by a1saipavan** [comment]
I'm able to do with AWS cli but unable to deploy with GitHub actions.

**Comment by vitalyk-multinarity** [comment]
If you're using Github managed runner, they cannot connect to the private address of your RDS.
There are at least two solutions:
- SSH to and un SQL from some internal EC2/container
- use self-hosted runner in your AWS network

**Comment by a1saipavan** [comment]
Can you please elaborate how can I create,
Please give me step by step flow, I'm new to this GitHub actions.

**Comment by vilmarsitiodigital** [comment]
hello,

I would try to solve this problem in two ways:

1 - Go to RDS > Databases and in the Connectivity tab expand Additional Configuration and check Publicly Accessible

2 - In this same part to improve security I would remove port 5432 and put a less usual port like 4325 just to avoid future attacks

3 - In EC2
Security groups enter related security group in your RDS and add above mentioned port in my case 4325

4 - With public access, redo the action

**Reply by Oatelaus** [reply]
This works but is degrading security; port enumeration would quickly surface your service.

Your database should never be publicly accessible.
